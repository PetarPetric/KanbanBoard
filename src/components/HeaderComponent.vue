<template>
  <header class="header">
    <img class="header__logo" src="@/assets/random-logo.png" alt="application logo">
    <h1>Kanban Board</h1>
  </header>
</template>

<style lang="scss" scoped>
.header {
  height: 5.6rem;
  border-bottom: 1px solid #e8ebed;
  padding: 1rem;
  display: flex;
  align-items: center;

  &>* {
    padding: 0 1rem;
  }

  h1 {
    font-size: 2rem;
  }

  &__logo {
    height: 100%;
  }
}
</style>
