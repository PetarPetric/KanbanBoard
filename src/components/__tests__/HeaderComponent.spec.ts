import { mount, VueWrapper } from "@vue/test-utils";
import HeaderComponent from "@/components/HeaderComponent.vue";

describe("header component", () => {
  let wrapper: VueWrapper<any>;

  beforeEach(() => {
    wrapper = mount(HeaderComponent, {
      shallow: true,
    });
  });

  // Simple test for header component
  
  it("should render correctly", () => {
    expect(wrapper.text()).toContain("Kanban Board");
  });

  it("should match snapshot", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
