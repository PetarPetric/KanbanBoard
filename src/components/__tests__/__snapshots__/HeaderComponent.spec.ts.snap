// Vitest Snapshot v1

exports[`header component > should match snapshot 1`] = `
"<header class=\\"header\\" data-v-2c8eaea2=\\"\\"><img class=\\"header__logo\\" src=\\"/assets/random-logo.png\\" alt=\\"application logo\\" data-v-2c8eaea2=\\"\\">
  <h1 data-v-2c8eaea2=\\"\\">Kanban Board</h1>
</header>"
`;
